### ISSUE TITLE
Bug: Missing errors in space-in-parens with nested parens

### ISSUE THREAD

**Issue by alberto** [eslint/eslint#4257]
In `space-in-parens`, with options: `never`

I would expect this code to produce 4 errors, but it produces just 2:
```js
( ( 1 + 2 ) )
```

For reference, this correctly produces 4 errors
```js
(  3 * ( 1 + 2 ) + 4 )
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
My hunch is that we aren't seeing the double parentheses because it doesn't alter the AST at all. The AST with single, double, and no parentheses ends up the same.

**Comment by nzakas** [comment]
Which is to say, this is a bug, but I'm not sure how to fix it.

**Comment by alberto** [comment]
Right now it is text based, although I intend to rewrite it with tokens if @mysticatea is not working already on it (from his comment #3996). I think it can be fixed using tokens.

Is it ok to rewrite the rule as part of the fix or should I also create a new issue for that?

**Comment by nzakas** [comment]
It's fine to do it as part of this.

**Comment by alberto** [comment]
Working on this.
